From 52806a3cec9c905ab4d324e12850bfd433a002d8 Mon Sep 17 00:00:00 2001
From: ilya <lystsov.ilya@gmail.com>
Date: Sun, 24 Mar 2024 23:52:51 +0300
Subject: [PATCH 07/13] add hitpoints in sendall

---
 20240318/1/client.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/20240318/1/client.py b/20240318/1/client.py
index dd22145..1e1590f 100644
--- a/20240318/1/client.py
+++ b/20240318/1/client.py
@@ -110,7 +110,7 @@ class MultiUserDungeonShell(cmd.Cmd):
             print("Cannot add unknown monster")
             return
 
-        self.socket.sendall(f"addmon {name} {x} {y} {greetings_message}".encode())
+        self.socket.sendall(f"addmon {name} {hitpoints} {x} {y} {greetings_message}".encode())
         print(f"Added monster {name} to ({x}, {y}) saying {greetings_message}")
         server_response = shlex.split(self.socket.recv(1024).decode())
         replaced_flag = server_response
-- 
2.34.1

