From 4e8ee26735d8c30f32d08f7f9eca79c122ceba21 Mon Sep 17 00:00:00 2001
From: ilya <lystsov.ilya@gmail.com>
Date: Mon, 25 Mar 2024 00:43:59 +0300
Subject: [PATCH 12/13] fix returning types in move function

---
 20240318/1/server.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/20240318/1/server.py b/20240318/1/server.py
index 87883e7..d2f0626 100644
--- a/20240318/1/server.py
+++ b/20240318/1/server.py
@@ -29,14 +29,14 @@ class MultiUserDungeon:
         name = monster.name
         return name, text
 
-    def move_player(self, delta_x, delta_y):
+    def move_player(self, delta_x, delta_y) -> tuple[str, str, str, str, str]:
         self.player.x = (self.player.x + delta_x) % self.field_size
         self.player.y = (self.player.y + delta_y) % self.field_size
         if (self.player.x, self.player.y) in self.monsters:
             name, text = self.encounter(self.player.x, self.player.y)
-            return self.player.x, self.player.y, "True", name, text
+            return str(self.player.x), str(self.player.y), "True", name, text
         else:
-            return self.player.x, self.player.y, "False", '', ''
+            return str(self.player.x), str(self.player.y), "False", '', ''
 
     def add_monster(self, name, text, hp, x, y):
         if (x, y) in self.monsters:
-- 
2.34.1

